client/ayon_blender/plugins/load/load_blend_link.py

```diff
@@ -0,0 +1,209 @@
+        new_filepath = self.filepath_from_context(context)
+        # Update library filepath and reload it
+        new_filename = os.path.basename(new_filepath)
+        library = self._get_or_build_library_by_path(new_filepath)
+        library.name = new_filename
+        library.filepath = new_filepath
+        library.reload()
+
+        # refresh UI
+        bpy.context.view_layer.update()
+        # Update container metadata
+        updated_data = {
+            "representation": str(repre["id"]),
+            "libpath": new_filepath
+        }
+        metadata_update(collection, updated_data)
+        self._update_library_path_by_node(collection, new_filepath)
+
+    def exec_remove(self, container: Dict) -> bool:
+        """Remove existing container from the Blender scene."""
+        collection = container["node"]
+        library = self._get_library_from_collection(collection)
+        if library:
+            linked_to_coll = self._has_linked_to_existing_collections(library)
+            if not linked_to_coll:
+                bpy.data.libraries.remove(library)
+        else:
+            # Ensure the collection is linked to the scene's master collection
+            scene_collection = bpy.context.scene.collection
+            for col in collection.children:
+                scene_collection.children.unlink(col)
+        # remove the container collection
+        bpy.data.collections.remove(collection)
+
+        return True
+
+    def _get_library_from_collection(
+            self, collection: bpy.types.Collection) -> bpy.types.Library:
+        """Get the library from the collection."""
+        for child in collection.children:
+            if child.library:
+                # No override library
+                return child.library
+            # With override library
+            elif child.override_library and child.override_library.reference:
+                return child.override_library.reference.library
+
+        return None
+
+    def _has_linked_to_existing_collections(
+            self, library: bpy.types.Library) -> bool:
+        """Check if any collection loaded by link
+        scene loader linked to the same library.
+        """
+        existing_collections = [
+            container["node"] for container in ls()
+            if container["loader"] == str(
+                self.__class__.__name__)
+        ]
+        match_count = list(
+            coll for coll in existing_collections
+            if self._get_library_from_collection(coll) == library
+        )
+
+        return len(match_count) > 1
+
+    def _get_or_build_library_by_path(self, libpath: str) -> bpy.types.Library:
+        """Get the library by filepath. If there is no any
+        associated library to the path, the related library is loaded
+        accordingly."""
+        lib_name = os.path.basename(libpath)
+        for library in bpy.data.libraries:
+            if lib_name == library.name_full:
+                return library
```

Comment:
Why aren't we just comparing by the [library `filepath`](https://docs.blender.org/api/current/bpy.types.Library.html#bpy.types.Library.filepath)?


So that we just check - is this file already loaded? Yes, use it... otherwise load new one.